docs/whirlpool/docs/02-Architecture Overview/07-TokenExtensions Support.md

```diff
@@ -0,0 +1,161 @@
+|initialize_reward_v2|initialize_reward|
+
+### ConfigExtension & TokenBadge Instructions
+- initialize_config_extension
+- set_config_extension_authority
+- set_token_badge_authority
+- initialize_token_badge
+- delete_token_badge
+
+## Token Badge
+It was determined that the ability to freely create pools with tokens that have functions that may malfunction the pools, such as PermanentDelegate, has more disadvantages for abuse than advantages. However, several stable coins already have PermanentDelegate extension.
+
+If tokens with features such as PermanentDelegate are used to create pools or initialize rewards, some whitelist mechanism is required, which is TokenBadge.
+
+TokenBadge is a PDA whose seed is WhirlpoolsConfig and Mint, and each WhirlpoolsConfig can control whether or not to create TokenBadge in its space.
+
+The TokenBadge itself only records the WhirlpoolsConfig and Mint used at initialization, and has no other additional information at the moment.
+
+![TokenBadge](../../static/img/02-Architecture%20Overview/token-badge.png)
+
+## Notes
+
+### Freeze Authority
+Token-2022 tokens with freeze authority will be rejected unless TokenBadge account is initialized.
+
+If you initialise the pool with your Token-2022 tokens, make sure you have disabled FreezeAuthority.
+
+### Native Token (WSOL)
+Token-2022 program has its own Wrapped SOL mint address (9pan9bMn5HatX4EJdBwg9VgCa7Uz5HL8N1m5D3NdXejP). Let's call it WSOL-2022.
+
+Whirlpool program rejects WSOL-2022.
+
+WSOL-2022 has no token extensions, so please use original WSOL (So11111111111111111111111111111111111111112).
+
+### TransferFee
+Anyone can initialize whirlpool with Token-2022 tokens having `TransferFeeConfig` extension.
+
+#### `amount` and `otherAmountThreshold`
+The relationship between `amount`, `otherAmountThreshold` and `TransferFee` in each context.
+
+The relationship between amount, otherAmountThreshold and TransferFee in each context.
+
+ExactIn (`amount_specified_is_input = true`)
+- `amount`: transfer fee Included amount (will be sent from user's token account)
+- `otherAmountThreshold`: transfer fee Excluded amount (will be received at user's token account)
+
+ExactOut (`amount_specified_is_input = false`)
+- `amount`: transfer fee Excluded amount (will be received at user's token account)
+- `otherAmountThreshold`: transfer fee Included amount (will be sent from user's token account)
+
+The user can set conditions on the amount actually going out of and coming into the token account, regardless of the amount of fees.
+
+No additional parameters are added to limit the amount of fees.
+If the fee will be changed, it is forced to be activated two epochs after, so there is no possibility of the fee suddenly going up. The edge case is when the fee change is scheduled (new epoch is coming) in less than transaction life time. In this case, the UI may need to alert users, but in any case, transactions that exceed the outgoing and incoming amount thresholds will fail.
```

Comment:
> In this case, the UI may need to alert users, but in any case, transactions that exceed the outgoing and incoming amount thresholds will fail.

Not sure how relevant this comment is. What UI are we talking about? Maybe we should just say that a transaction in this case would fail or something like that